ruby/ext/google/protobuf_c/storage.c

```diff
@@ -178,9 +178,42 @@ void native_slot_set_value_and_case(const char* name,
       if (CLASS_OF(value) == CLASS_OF(Qnil)) {
         value = Qnil;
       } else if (CLASS_OF(value) != type_class) {
-        rb_raise(cTypeError,
-                 "Invalid type %s to assign to submessage field '%s'.",
-                rb_class2name(CLASS_OF(value)), name);
+        // check for possible implicit conversions
+        VALUE converted_value = NULL;
+        char* field_type_name = rb_class2name(type_class);
+        char* value_type_name = rb_class2name(CLASS_OF(value));
+
+        // Time -> Google::Protobuf::Timestamp
+        if (strcmp(field_type_name, "Google::Protobuf::Timestamp") == 0 &&
```

Comment:
upb provides upb_msgdef_wellknowntype. You don't need strcmp here.

Reply:
Hmm... I couldn't find too many usages of that method. How can I get the `upb_msgdef` from the `type_class` in order to use it?

Reply:
VALUE value;
MessageHeader* msg
TypedData_Get_Struct(value, MessageHeader, &Message_type, msg);
msg->descriptor->msgdef

Reply:
ok.. seems you cannot get msgdef from type_class